### PR TITLE
PLAT-786: Add event search API to spec

### PR DESCRIPTION
**Pull request by srri**
This PR adds the API definition for the new event search API https://github.com/fingerprintjs/koala/pull/7030
This is a public facing API that will return a list of the existing APIv3 event format.

This API can filter on visitor id, ip address/range, suspect flag, bot type, time ranges, and linked id.

This is also my first change directly to our open API repo, so let me know if I missed something (for example, a version bump)

cc: @petrpalata

### CODE REVIEW

**Comment by srri** [comment]
I'm sure I've probably missed something, but setting for review.

File: schemas/components/schemas/Identification.yaml [hunk 1]
```diff
@@ -33,6 +33,9 @@ properties:
   linkedId:
     type: string
     description: A customer-provided id that was sent with the request.
+  suspect:
```
Comment (srri) [inline schemas/components/schemas/Identification.yaml:36]: we are adding this field, previously defined and released in the update API, to the top level of an identification (similar to tag, linkedId, being customer provided).

Comment (JuroUhlar) [reply schemas/components/schemas/Identification.yaml:36]: Can confirm it shows up in the regular get event endpoint response too

**Comment by ilfa** [comment]
@srri could you add release notes to this PR as described [here](https://github.com/fingerprintjs/fingerprint-pro-server-api-openapi/blob/main/CONTRIBUTING.md#describing-changes)?

File: schemas/paths/event-search.yaml [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    Please note that the response includes mobile signals (e.g. `rootApps`) even if the request originated from a non-mobile platform.
+    It is highly recommended that you **ignore** the mobile signals for such requests. 
+    If a `start` or `end` time are not supplied, the default search range is the last 7 days.
+  parameters:
+    - name: visitor_id
+      in: query
+      schema:
+        type: string
+      description: | 
+        Unique [visitor identifier](https://dev.fingerprint.com/reference/get-function#visitorid) issued by Fingerprint Pro.
```
Comment (JuroUhlar) [inline schemas/paths/event-search.yaml:17]:
```suggestion
        Unique [visitor identifier](https://dev.fingerprint.com/reference/get-function#visitorid) issued by Fingerprint.
```

File: schemas/paths/event-search.yaml [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+          - all
+          - good
+          - bad
+          - none
+      description: |
+        Filter events by the classification of bot type, specifically: 
+          - all events classified as any kind of bot.
+          - events detected classified as a good bot.
+          - events detected classified as a bad bot.
+          - only non-bot events.
```
Comment (JuroUhlar) [inline schemas/paths/event-search.yaml:33]:
```suggestion
        Filter events by the bot detection result, specifically: 
          - events where any kind of bot was detected.
          - events where a good bot was detected.
          - events where a bad bot was detected.
          - events where no bot was detected.
```

File: schemas/paths/event-search.yaml [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    - name: ip_address
+      in: query
+      schema:
+        type: string
+      description: |
+        Filter events by IP address range. Range can be as specific as a single IP (/32 for IPv4 or /128 for IPv6) 
+        All ip_address filters must use CIDR notation, for example: 10.0.0.0/24, 192.168.0.1/32
```
Comment (JuroUhlar) [inline schemas/paths/event-search.yaml:40]:
```suggestion
        Filter events by IP address range. The range can be as specific as a single IP (/32 for IPv4 or /128 for IPv6) 
        All ip_address filters must use CIDR notation, for example, 10.0.0.0/24, 192.168.0.1/32
```

File: schemas/paths/event-search.yaml [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+    - name: linked_id
+      in: query
+      schema:
+        type: string
+      description: |
+        Filter events by your custom identifier. 
+        
+        You can use [`linked_id`](https://dev.fingerprint.com/reference/get-function#linkedid) to associate identification requests with your own identifier, for example: session ID, purchase ID, or transaction ID. You can then use this `linked_id` parameter to retrieve all events associated with your custom identifier.
```
Comment (JuroUhlar) [inline schemas/paths/event-search.yaml:48]:
```suggestion
        You can use [linked IDs](https://dev.fingerprint.com/reference/get-function#linkedid) to associate identification requests with your own identifier, for example, session ID, purchase ID, or transaction ID. You can then use this `linked_id` parameter to retrieve all events associated with your custom identifier.
```

File: schemas/paths/event-search.yaml [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+        format: int64
+      description: |
+        Filter events with a timestamp greater than the start time, in unix time (milliseconds).
+    - name: end
+      in: query
+      schema:
+        type: integer
+        format: int64
+      description: |
+        Filter events with a timestamp less than the end time, in unix time (milliseconds).
```
Comment (JuroUhlar) [inline schemas/paths/event-search.yaml:72]:
```suggestion
        Filter events with a timestamp smaller than the end time, in Unix time (milliseconds).
```

File: schemas/paths/event-search.yaml [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+        minimum: 1
+      description: |
+        Limit the number of events returned.
+    - name: start
+      in: query
+      schema:
+        type: integer
+        format: int64
+      description: |
+        Filter events with a timestamp greater than the start time, in unix time (milliseconds).
```
Comment (JuroUhlar) [inline schemas/paths/event-search.yaml:65]:
```suggestion
        Filter events with a timestamp greater than the start time, in Unix time (milliseconds).
```

File: schemas/paths/event-search.yaml [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+            $ref: '../components/schemas/ErrorResponse.yaml'
+          examples:
+            403-token-required:
+              summary: Error response when the secret API key was not provided.
+              externalValue: 'examples/errors/403_token_required.json'
+            403-token-not-found:
+              summary: Error response when the provided secret API key does not exist.
+              externalValue: 'examples/errors/403_token_not_found.json'
+            403-wrong-region:
+              summary: Error response when the API region is different from the region, the calling application is configured with.
```
Comment (JuroUhlar) [inline schemas/paths/event-search.yaml:110]:
```suggestion
              summary: Error response when the API region does not match the region of your Fingerprint workspace.
```

File: schemas/paths/event-search.yaml [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+            $ref: '../components/schemas/ErrorResponse.yaml'
+          examples:
+            400-limit-invalid:
+              summary: Error response when no limit is supplied, or is invalid.
+              externalValue: 'examples/errors/400_limit_invalid.json'
+            400-ip-address-invalid:
+              summary: Error response when an invalid IP address is supplied, or is not using CIDR notation.
+              externalValue: 'examples/errors/400_ip_address_invalid.json'
+            400-bot-type-invalid:
+              summary: Error response when an invalid bot type is specified, must be one of good, bad, all, none.
```
Comment (JuroUhlar) [inline schemas/paths/event-search.yaml:126]:
```suggestion
              summary: Error response when an invalid bot type is specified, must be one of `good`, `bad`, `all`, or `none`.
```

File: schemas/paths/event-search.yaml [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+              summary: Error response when the reverse parameter is invalid.
+              externalValue: 'examples/errors/400_reverse_invalid.json'
+            400-start-time-invalid:
+              summary: Error response when an invalid start time is supplied.
+              externalValue: 'examples/errors/400_start_time_invalid.json'
+            400-end-time-invalid:
+              summary: Error response when an invalid end time is supplied.
+              externalValue: 'examples/errors/400_end_time_invalid.json'
+            400-visitor-id-invalid:
+              summary: Error response when an invalid visitor id is supplied.
```
Comment (JuroUhlar) [inline schemas/paths/event-search.yaml:138]:
```suggestion
              summary: Error response when an invalid visitor ID is supplied.
```

File: schemas/paths/event-search.yaml [hunk 11]
```diff
@@ -0,0 +1,145 @@
...
+              externalValue: 'examples/errors/400_visitor_id_invalid.json'
+            400-linked-id-invalid:
+              summary: Error response when an invalid linked id is supplied (too large).
```
Comment (JuroUhlar) [inline schemas/paths/event-search.yaml:141]:
```suggestion
              summary: Error response when an invalid (too large) linked ID is supplied.
```

**Review by JuroUhlar — CHANGES_REQUESTED** [review]

**Review by JuroUhlar — DISMISSED** [review]

**Review by JuroUhlar — APPROVED** [review]

**Review by JuroUhlar — APPROVED** [review]

**Pull request merged**